File: src/main/java/kr/hahaha98757/zombiesaddon/features/TextMacro.java
package kr.hahaha98757.zombiesaddon.features;

import kr.hahaha98757.zombiesaddon.KeyBindings;
import kr.hahaha98757.zombiesaddon.config.ZombiesAddonConfig;
import kr.hahaha98757.zombiesaddon.utils.Utils;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.InputEvent;

public class TextMacro {

    @SubscribeEvent
    public void onKeyInput(InputEvent.KeyInputEvent event) {
        if (Utils.isModDisable()) return;

        if (!KeyBindings.textMacro.isPressed()) return;

        Utils.sendChat(ZombiesAddonConfig.getTextMacro());
    }
}

File: src/main/kotlin/kr/hahaha98757/zombiesaddon/modules/TextMacro.kt
package kr.hahaha98757.zombiesaddon.modules

import kr.hahaha98757.zombiesaddon.KeyBindings
import kr.hahaha98757.zombiesaddon.config.ZAConfig
import kr.hahaha98757.zombiesaddon.utils.isDisable
import kr.hahaha98757.zombiesaddon.utils.isNotPlayZombies
import kr.hahaha98757.zombiesaddon.utils.sendChat
import net.minecraftforge.fml.common.gameevent.InputEvent

object TextMacro: AlwaysEnableModule("Text Macro") {
    override fun onKeyInput(event: InputEvent.KeyInputEvent) {
        if (!KeyBindings.textMacro.isPressed) return
        if (isDisable()) return
        if (isNotPlayZombies()) return
        sendChat(ZAConfig.textMacro)
    }
}